src/env_node.c

```diff
@@ -11,8 +11,8 @@ t_env_node	*env_node_create_from_pair(t_pair *pair, t_scope_e scope)
 	if (node)
 	{
 		node->environ_index = -1;
-		node->key = pair->f.key;
-		node->value = pair->s.value;
+		node->key = pair->u_f.key;
+		node->value = pair->u_s.value;
```

Comment:
you should checkout the anonymous union 
https://www.geeksforgeeks.org/g-fact-38-anonymous-union-and-structure/
you can have
	union
	{
		void	*first;
		char	*key;
	};
	union
	{
		void	*second;
		void	*value;
	};

and acces by pair->key